openssh/files/default/ssh_known_hosts

```diff
@@ -81,26 +81,19 @@
 {%- set host_names = salt['mine.get'](target, hostname_function, tgt_type=tgt_type) -%}
 
 {#- Salt SSH (if any) #}
-{%- for minion_id, minion_host_keys in salt['pillar.get'](
-  'openssh:known_hosts:salt_ssh:public_ssh_host_keys',
-  {}
-).items() -%}
+{%- for minion_id, minion_host_keys in known_hosts | traverse('salt_ssh:public_ssh_host_keys', {}) | dictsort -%}
 {%-   if salt["match.{}".format(tgt_type)](target, minion_id=minion_id) -%}
 {%      do host_keys.update({minion_id: minion_host_keys}) %}
 {%-   endif -%}
 {%- endfor -%}
-{%- for minion_id, minion_host_names in salt['pillar.get'](
-  'openssh:known_hosts:salt_ssh:public_ssh_host_names',
-  {}
-).items() -%}
+{%- for minion_id, minion_host_names in known_hosts | traverse('salt_ssh:public_ssh_host_names', {}) | dictsort -%}
 {%-   if salt["match.{}".format(tgt_type)](target, minion_id=minion_id) -%}
 {%      do host_names.update({minion_id: minion_host_names}) %}
 {%-   endif -%}
 {%- endfor %}
 
 {#- Static Pillar data #}
-{%- do host_keys.update(salt['pillar.get']('openssh:known_hosts:static',
-    {}).items()) -%}
+{%- do host_keys.update(known_hosts | traverse('static', {})) -%}
```

Comment:
The `.items()` has been removed here -- was that intentional?

Reply:
I saw a dict `.update` and thought that the `.items` where superfluous. I must make tests to see what the `.items` did exactly.

Nice catch.

Reply:
I did a little test:

```jinja
{%- set h_w_items = {'a': 1, 'b': 2} %}
{%- set h_wo_items = {'a': 1, 'b': 2} %}

{%- set h2 = {'c': 3, 'd': 4} %}

{%- do h_w_items.update(h2.items()) -%}
{%- do h_wo_items.update(h2) -%}

# With items
{{ h_w_items }}

# Without items
{{ h_wo_items }}
```

Results in the followning:

```
# With items
{'a': 1, 'b': 2, 'c': 3, 'd': 4}

# Without items
{'a': 1, 'b': 2, 'c': 3, 'd': 4}
```

Which seems consistent with [python documentation](https://docs.python.org/3/library/stdtypes.html#dict.update):

> update() accepts either another dictionary object or an iterable of key/value pairs (as tuples or other iterables of length two). If keyword arguments are specified, the dictionary is then updated with those key/value pairs: d.update(red=1, blue=2).

Reply:
I just noticed that there is [a final `| dictsort` on the `host_keys`](https://github.com/saltstack-formulas/openssh-formula/blob/ea221ab52b0bd77173e83f5eb8b116324ad7c280/openssh/files/default/ssh_known_hosts#L106).

I will use intermediate variables to avoid many calls to `dictsort` and opened a [feature request to salt](https://github.com/saltstack/salt/issues/57970) to have a new `| items` filter.

Reply:
The call to `.items()` was useless in the context of `.update()`.

Thanks.